### PR TITLE
DM-31616: copy code and tests from ts_salobj

### PR DESCRIPTION
**Pull request by r-owen**
and make the tests more pytest-like. I still use unittest.TestCase for the sake of subTest (because I don't like the pytest equivalent as well) but could be convinced otherwise.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        work_branches = "${GIT_BRANCH} ${CHANGE_BRANCH} develop"
+        LSST_IO_CREDS = credentials("lsst-io")
+        SQUASH_CREDS = credentials("squash")
+    }
+
+    stages {
+        stage("Pulling docker image") {
+            steps {
+                script {
+                    sh "docker pull lsstts/salobj:develop"
```
Comment (wvreeven) [inline Jenkinsfile:27]: The `lsstts/develop-env:develop` docker container should be used.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                }
+            }
+        }
+        stage("Preparing environment") {
+            steps {
+                script {
+                    sh """
+                    docker network create \${network_name}
+                    chmod -R a+rw \${WORKSPACE}
+                    container=\$(docker run -v \${WORKSPACE}:/home/saluser/repo/ -td --rm --net \${network_name} -e LTD_USERNAME=\${LSST_IO_CREDS_USR} -e LTD_PASSWORD=\${LSST_IO_CREDS_PSW} --name \${container_name} lsstts/salobj:develop)
```
Comment (wvreeven) [inline Jenkinsfile:37]: The `lsstts/develop-env:develop` docker container should be used.

**Review by wvreeven — APPROVED** [review]
In principles this looks good. Please fix the headers of three Python files which are referring to ts_salobj. Also please make sure to use the correct docker container for the Jenkins CI build.

File: Jenkinsfile (hunk 2 above)
Comment (r-owen) [reply Jenkinsfile:37]: Why? I would expect something even simpler than salobj would do (if we have something simpler) as this package doesn't do anything with dds (including ts_sal, ts_salobj, IDL files, etc.) or configuration (ts_config_x). @tribeiro do you have an opinion on the best docker image to use?

Comment (wvreeven) [reply Jenkinsfile:37]: I asked Tiago about this on June 9th this year on Slack. I asked `What would be the recommended docker image to use in Jenkinsfile? lsstts/salobj or lsstts/develop-env?` and his answer was `Second one`.

File: Jenkinsfile (hunk 1 above)
Comment (tribeiro) [reply Jenkinsfile:27]: I would suggest actually using the salobj image.. the develop-env image is useful for CSCs and other components that need more of the development environment.. The salobj images is a precursor of develop-env which can be used for packages that has less dependencies..

File: Jenkinsfile (hunk 2 above)
Comment (tribeiro) [reply Jenkinsfile:37]: For CSCs and things that depends on more packages, I'd go for develop-env.. This package is a good candidate to use the simpler salobj one.

**Pull request merged**